forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -421,8 +421,8 @@ public static int calculateAmount(final Card card, String amount, final CardTrai
         // return empty strings and constants
         if (StringUtils.isBlank(amount)) { return 0; }
         if (card == null) { return 0; }
-        final Player player = card.getController();
-        final Game game = player == null ? card.getGame() : player.getGame();
+        final Player player = ability instanceof SpellAbility ? ((SpellAbility)ability).getActivatingPlayer() : card.getController();
+        final Game game = card.getGame();
```

Comment:
what if player is null? it should never ever be null, right?
in case the CardTrait is not a SpellAbility, and card.getController is currently null?

i think that might happen for debug or some other corner cases?

Reply:
hmn I guess a fallback to card doesn't hurt